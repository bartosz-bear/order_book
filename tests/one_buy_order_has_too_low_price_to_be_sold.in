{"type": "Limit", "order": {"direction": "Buy", "id": 1, "price": 14, "quantity": 20}}
{"type": "Iceberg", "order": {"direction": "Buy", "id": 2, "price": 16, "quantity": 20, "peak": 3}}
{"type": "Iceberg", "order": {"direction": "Buy", "id": 3, "price": 15, "quantity": 20, "peak": 3}}
{"type": "Limit", "order": {"direction": "Sell", "id": 4, "price": 15, "quantity": 39}}