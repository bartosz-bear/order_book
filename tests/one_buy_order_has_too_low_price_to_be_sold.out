{"buyOrders": [{"id": 1, "price": 14, "quantity": 20}], "sellOrders": []}
{"buyOrders": [{"id": 2, "price": 16, "quantity": 3}, {"id": 1, "price": 14, "quantity": 20}], "sellOrders": []}
{"buyOrders": [{"id": 2, "price": 16, "quantity": 3}, {"id": 3, "price": 15, "quantity": 3}, {"id": 1, "price": 14, "quantity": 20}], "sellOrders": []}
{"buyOrders": [{"id": 3, "price": 15, "quantity": 1}, {"id": 1, "price": 14, "quantity": 20}], "sellOrders": []}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 16}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 16}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 16}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 16}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 16}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 3, "price": 16}
{"buyOrderId": 2, "sellOrderId": 4, "quantity": 2, "price": 16}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 15}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 15}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 15}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 15}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 15}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 3, "price": 15}
{"buyOrderId": 3, "sellOrderId": 4, "quantity": 1, "price": 15}